src/praisonai-agents/praisonaiagents/tools/advanced_tools.py

```
@@ -0,0 +1,492 @@
+"""Advanced tools framework for PraisonAI Agents.
+
+This module provides advanced tool decorators and functionality including:
+- Pre/Post execution hooks
+- Tool-level caching with TTL
+- External execution markers
+- Structured user input fields
+
+Maintains backward compatibility with existing tools.
+"""
+
+import asyncio
```

Ruff (0.12.2)

`asyncio` imported but unused

Remove unused import: `asyncio`

(F401)